### ISSUE TITLE
Javascript Functions should default to "cardinality": "many" for EventHubs

### ISSUE THREAD

**Issue by mhoeger** [Azure/azure-functions-host#2815]
The default behavior that we recommend is to batch trigger events. However, we have it as an opt-in setting right now. We should change this so that best practice = default behavior

https://stackoverflow.com/questions/46204764/eventhub-trigger-to-functions-cardinality-with-one-causing-higher-timeouts

**Comment by paulbatum** [comment]
So I think our template already defaults to many. I assume this issue is tracking that in terms of function.json, you don't need to say "cardinality:many"?

**Comment by mhoeger** [comment]
Yep! The change is to default to batching if "cardinality" is omitted.

**Comment by mhoeger** [comment]
Update: the way that "cardinality": "many" works is to transform _any and all input data_ to an array type if cardinality == many on a given function. We can't default cardinality to many without a serious re-write of where this logic goes.

There is a bug where dataType had to be specified to "string" on all event hubs trigger data that is not valid json, which is a regression from v1. 

The fix is here: https://github.com/Azure/azure-functions-host/pull/3170
To address this, will also make documentation more clear that "cardinality": "many" is best practice

**Comment by mhoeger** [comment]
Moving to backlog as a "nice to have", but not practical.

**Comment by brettsam** [comment]
>Update: the way that "cardinality": "many" works is to transform any and all input data to an array type if cardinality == many on a given function. We can't default cardinality to many without a serious re-write of where this logic goes.

Judging by this comment, it seems like this is something we probably won't be doing and instead make sure templates/docs explain that this is best practices. 

@pragnagopa / @fabiocav -- do either of you have context on this one?

**Comment by paulbatum** [comment]
I believe the related issues spell out a strategy for fixing this. Having docs that explain that batched processing is a best practice doesnt help if the customer can't use it because their data is in a non-JSON format (see https://github.com/Azure/azure-functions-eventhubs-extension/issues/118). So from my perspective this is work that should happen, and its mostly a matter of prioritization.
